### PR TITLE
Nearest Grid Point Interpolation

### PR DESCRIPTION
**Pull request by sslattery**
Adds nearest grid point interpolation via a zero-order spline.

### CODE REVIEW

**Comment by sslattery** [comment]
> This looks reasonable to me, but I'm not feeling confident enough with the over arching machinery to rubber stamp it

Basically you just dump all your data to the nearest node, cell, edge, face, etc. that you have specified. The gradient is not available as the shape function is just a constant. This is good for implementing older methods and simple things like counting how many particles are in a cell.

**Comment by streeve** [comment]
Can you restart the CI?

File: src/Cajita_Splines.hpp [hunk 1]
```diff
@@ -14,6 +14,93 @@ namespace Cajita
...
+    /*!
+      \brief Compute the stencil indices for a given logical space location.
+      \param x0 The coordinate at which to evaluate the spline stencil.
+      \param indices The indices of the stencil.
+    */
+    template <class Scalar>
+    KOKKOS_INLINE_FUNCTION static void stencil( const Scalar x0,
+                                                int indices[num_knot] )
+    {
+        indices[0] = int( x0 );
```
Comment (dalg24) [inline src/Cajita_Splines.hpp]: Avoid functional cast expression which is equivalent to the C-style cast `(int)x0`.
Prefer `static_cast<int>(x0)`

**Comment by sslattery** [comment]
> Is that really nearest? You are always rounding down aren't you?

Yes it is nearest - note that this (and all even-order interpolations) are defined on the dual grid. In 1D, the primal grid has "cells" which are bounding by the grid nodes and the dual grid has "cells" which are bounding by the cell centers of the primal grid. In the `mapToLogicalGrid` function there is an addition `0.5` which is used to map to the dual grid.

So, consider a mesh with the first node starting at `0.0` and a cell size of `1.0`. The node locations are then `0.0`, `1.0`, `2.0`, `3.0`, etc. and the cell centers are at `0.5`, `1.5`, etc. 

Now consider two points, one at `2.000001` and one at `2.9999999`. In the `mapToLogicalGrid` there is a parameter `low_x` which is the local low corner of the grid to which we are interpolating. If we are writing to nodes this is the left-most node location in the dimension and if we are writing to cells this is the left-most cell location in the dimension. The logical mapping of the first point is then `(2.000001 - 0.5) * 1.0 + 0.5 = 2.000001` and `(2.999999 - 0.5) * 1.0 + 0.5 = 2.999999` is the logical mapping of the second point. Round down to the nearest integer and both give a stencil index of 2 - therefore both interpolate to the cell with index 2 and of course both are in cell 2, hence nearest grid point interpolation.

The same is true for nodes where any point +/- a half cell width in either direction from the node will write to that node.

File: src/Cajita_Splines.hpp (hunk 1 above)
Comment (sslattery) [reply src/Cajita_Splines.hpp]: good point. I will fix

**Comment by sslattery** [comment]
> Can you restart the CI?

Some times the GitHub travis CI doesn't seem to start. The latest push seems to have worked.

**Comment by sslattery** [comment]
@dalg24 @rfbird Any more review comments on this one?

**Comment by rfbird** [comment]
Nope, send it!

**Pull request merged**